packages/alfa-rules/src/sia-r113/rule.ts

```diff
@@ -50,96 +58,133 @@ export default Rule.Atomic.of<Page, Element>({
+        size: true,
+      }),
+    );
+
+  export const HasSufficientSpacing = (name: string, box: Rectangle) =>
     Ok.of(
-      WithBoundingBox.of("Target has sufficient size or spacing", name, box),
+      WithBoundingBox.of("Target has sufficient spacing", name, box, {
+        size: false,
+        spacing: true,
+      }),
     );
 
   export const HasInsufficientSizeAndSpacing = (name: string, box: Rectangle) =>
     Err.of(
-      WithBoundingBox.of("Target has insufficient size and spacing", name, box),
+      WithBoundingBox.of(
+        "Target has insufficient size and spacing",
+        name,
+        box,
+        { size: false, spacing: false },
+      ),
     );
-
-  export const IsUserAgentControlled = (name: string, box: Rectangle) =>
-    Ok.of(WithBoundingBox.of("Target is user agent controlled", name, box));
-}
-
-function hasSufficientSizeOrSpacing(
-  document: Document,
-  device: Device,
-): Predicate<Element> {
-  return or(
-    hasSufficientSize(24, device),
-    hasSufficientSpacing(document, device),
-  );
 }
 
-const spacingCache = Cache.empty<
+const undersizedCache = Cache.empty<
   Document,
-  Cache<Device, Cache<Element, boolean>>
+  Cache<Device, Sequence<Element>>
 >();
 
 /**
  * @remarks
- * Spacing is calculated by
- * 1. drawing a 24px diameter circle around the center of the bounding box of the target,
- * 2. checking if the circle intersects with the bounding box of any other target, or
- * 3. if the circle intersects with the 24px diameter circle of another undersized target.
+ * This predicate tests that the target has sufficient spacing around it.
+ * The spacing is calculated by drawing a circle around the center of the bounding box of the target of radius 12.
+ * The target is underspaced, if
+ * 1) the circle intersects with the bounding box of any other target, or
+ * 2) the distance between the center of the bounding box of the target
+ *    and the center of the bounding box of any other **undersized** target is less than 24.
  */
 function hasSufficientSpacing(
   document: Document,
   device: Device,
 ): Predicate<Element> {
   return (target) => {
-    // The target might have been cached while computing the spacing for another target
-    const cachedTarget = spacingCache
-      .get(document, Cache.empty)
-      .get(device, Cache.empty)
-      .get(target);
-
-    if (cachedTarget.isSome()) {
-      return cachedTarget.get();
-    }
-
     // Existence of a bounding box is guaranteed by applicability
-    const box = target.getBoundingBox(device).getUnsafe();
+    const targetRect = target.getBoundingBox(device).getUnsafe();
 
-    for (const otherTarget of targetsOfPointerEvents(document, device)) {
-      if (target === otherTarget) {
+    const undersizedTargets = undersizedCache
+      .get(document, Cache.empty)
+      .get(device, () =>
+        targetsOfPointerEvents(document, device).reject(
+          hasSufficientSize(24, device),
+        ),
+      );
```

Comment:
**Question:** I somehow feel that this "belongs" to inside the `evaluate` function of the rule, without being fully able to pinpoint a clear reason. Maybe this is because it reuses the targets and redoes the same test that part of the expectation does.
We could start the `evaluate` function by gathering all targets and then splitting them in two depending on size. OTOH, `hasSufficientSize` seems simple enough that re-computing it is simpler than a look-up (for deciding if the target has indeed sufficient size). For the undersized targets, they are used a lot and should indeed be stored.

Reply:
This was talked about out of band. We will revisit it when we need to make updates to the applicability of the rule.